src/plasma-cash/user.ts

```diff
@@ -83,28 +91,50 @@ export class User extends Entity {
     )
   }
 
-  static createUser(
+  static async createUser(
     wallet: ethers.Signer,
+    dappchainPrivateKey: string,
     plasmaAddress: string,
     dappchainEndpoint: string,
     eventsEndpoint: string,
     dbPath?: string,
     startBlock?: BN,
     chainId?: string
-  ): User {
+  ): Promise<User> {
     const database = new PlasmaDB(dbPath)
     const ethPlasmaClient = new EthereumPlasmaClient(wallet, plasmaAddress, eventsEndpoint)
     const writer = createJSONRPCClient({ protocols: [{ url: dappchainEndpoint + '/rpc' }] })
     const reader = createJSONRPCClient({ protocols: [{ url: dappchainEndpoint + '/query' }] })
     const dAppClient = new Client(chainId || 'default', writer, reader)
-    // TODO: Key should not be generated each time, user should provide their key, or it should be retrieved through some one way mapping
-    const privKey = CryptoUtils.generatePrivateKey()
+    let privKey
+    if (dappchainPrivateKey === '0x') {
```

Comment:
It's a bit confusing to use `0x` to indicate the lack of a base64 encoded string, should make the parameter `dappchainPrivateKey: string | null`.